### PR TITLE
chore(ubergen): exclude cx-api and cloud-assembly-schema

### PR DESCRIPTION
**Pull request by nija-at**
The cx-api and cloud-assembly-schema packages define protocol between
the CDK framework and the CDK CLI.

With v2, they need to be published as public packages. The CLI will consume
them as regular dependencies, and the framework package, aws-cdk-lib, will
have them bundled.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by eladb — CHANGES_REQUESTED** [review]
I think the correct way is to do both: bundle them in aws-cdk-lib _and_ publish them as independent modules so they can be consumed by the CLI (and anyone else).

I also think cx-api is a mess and we should look into deprecating it for v2.

@iliapolo thoughts?

**Comment by nija-at** [comment]
> bundle them in aws-cdk-lib and publish them as independent modules

Curious why it also needs to be bundled in aws-cdk-lib.

**Comment by eladb** [comment]
> Curious why it also needs to be bundled in aws-cdk-lib.

I think it might be okay if this was a regular dependency but I have a gut feeling that we need to only have a single non-bundled dependency in aws-cdk-lib (`constructs`). I think this deserves a face-to-face discussion though.

File: package.json [hunk 1]
```diff
@@ -95,6 +95,10 @@
       "aws-cdk-lib/semver/**",
```
Comment (eladb) [inline package.json:95]: remove `semver` and `jsonSchema` from here as well, right?

**Review by eladb — APPROVED** [review]

**Comment by nija-at** [comment]
The build is failing with the error - 

```
[2020-11-13T13:15:49.324] [ERROR] jsii/compiler - Type model errors prevented the JSII assembly from being created
error JSII3002: Type "@aws-cdk/cx-api.AssemblyBuildOptions" cannot be used as a base interface because it is not exported from @aws-cdk/cx-api
error JSII9002: Unable to resolve type "@aws-cdk/cx-api.AssemblyBuildOptions". It may be @internal or not exported from the module's entry point (as configured in "package.json" as "main").
error JSII9002: Unable to resolve type "@aws-cdk/cx-api.MetadataEntry". It may be @internal or not exported from the module's entry point (as configured in "package.json" as "main").
error JSII9002: Unable to resolve type "@aws-cdk/cx-api.MissingContext". It may be @internal or not exported from the module's entry point (as configured in "package.json" as "main").
```

The fix is likely to re-export the types from within the bundled dependencies as part of `monocdk` and `aws-cdk-lib`. 

@RomainMuller had some further thoughts on why he thinks bundling is a bad idea here. I'll let him chime here.

@eladb - given that the scope of this change is slowly expanding, would you be ok with changing directions for our alpha release, and making `cx-api` and `cloud-assembly-schema` regular dependencies of `monocdk`, `aws-cdk-lib` and `aws-cdk`? 
This will help clear out the v2 pipeline and get our first alpha release out. I can take a note to re-visit this before RC.

**Comment by RomainMuller** [comment]
So I basically disagree *bundling* literally ANYTHING is a long-term viable solution to any problem... 🤷🏻‍♂️
- If you bundle it AND vend another separate library... Then you might end up with both in the closure & a lot of confusion down the path.
- Yarn 2 does not support bundled dependencies, because they're gross and a source of problems. Yarn PnP is pretty darn awesome, but implies Yarn 2, meaning no bundled dependencies. If you use bundled dependencies, nobody can use Yarn PnP if they use your library.
- This means we get copies of libraries that people simply *cannot* control (it's the version shipped in and no other). This is also a longer term maintenance nightmare if you need to security patch, etc... Because there is no changing the contents of the bundled libraries out there...

If some contract must be shared between aws-cdk-lib and aws-cdk, then it must be done as:
- `aws-cdk` depends on `aws-cdk-lib` and consumes that
- `aws-cdk` and `aws-cdk-lib` share a reference to `@aws-cdk/toolkit-api` or something (which is published separately)
- `aws-cdk` and `aws-cdk-lib` both include the same submodule (i.e: they both have a copy of it which is "created" by our build tools, NOT by `npm` bundled dependencies)

**Comment by eladb** [comment]
@RomainMuller I agree that eventually cx-schema should just be a copy that we bundle in both modules, not npm bundled from.

But... what would you propose for non-jsii modules? How do you propose that we use them?

**Pull request closed without merging**